### PR TITLE
Check any permutation of --env prefix in tests

### PR DESCRIPTION
**Pull request by miha-plesko**
Capstan prepends environment variables to the boot command based on map[env_name]string where order is not always the same. Therefore we implement a specialized BootCmdEquals checker that compares obtained bootcmd string with given bootcmd base and environment map.

NOTE: This change is already included in https://github.com/mikelangelo-project/capstan/pull/37, but since it affects many lines, it will be better to merge it here and then rebase the PR regarding recursive run.yamls.

### CODE REVIEW

**Comment by miha-plesko** [comment]
I like your proposal because it brings many benefits. Most notably, besides that we won't be changing mainstream just for the sake of unit tests, we will be able to provide useful error message yelling about the problematic part of the boot command specifically. I will have to have the `DirEquals` checker extended to support test cases like [this one](https://github.com/mikelangelo-project/capstan/pull/37/files#diff-ab78a95482af62ccba054cd71487fddbR440), but this would sooner or later be the case.

Bottom line: I'm on it right away!

**Comment by miha-plesko** [comment]
There you go, I've pushed your proposal implemented as a `BootCmdEquals` checker. I decided not to go with regex at this point since we have no usecase for it, but if we ever need it, we can easily extend the checker into `BootCmdMatches`.

I kindly ask for a review.

**Comment by gberginc** [comment]
Please, update the PR title and text to match the new logic.

**Comment by miha-plesko** [comment]
Done.

File: test/runtime/runtime_test.go [hunk 1]
```diff
@@ -22,61 +23,49 @@ var _ = Suite(&testingRuntimeSuite{})
...
 			"",
 		},
 		{
 			"two variables in environment",
-			"/node server.js",
-			map[string]string{"PORT": "8000", "ENDPOINT": "foo.com"},
-			false,
-			// Order is not guaranteed hence the following regex is needed:
-			"(--env=PORT=8000 --env=ENDPOINT=foo.com|--env=ENDPOINT=foo.com --env=PORT=8000) /node server.js",
+			"/node server.js", map[string]string{"PORT": "8000", "ENDPOINT": "foo.com"}, false,
```
Comment (gberginc) [inline test/runtime/runtime_test.go]: If you use it like this, you could remove the `expectedEnv` and `expectedSoft` because they are always the same as those provided in `env` and `soft`.

I thought using explicit envs would be safer, i.e. 
```
	m := []struct {
		comment     string
		cmd         string
		env         map[string]string
		soft        bool
		expectedCmd string
		expectedEnv []string
		err         string
	}{
		{
			"two variables in environment - soft",
			"/node server.js",
			map[string]string{"PORT": "8000", "ENDPOINT": "foo.com"},
			true,
			// Order is not guaranteed hence the following regex is needed:
			"(--env=PORT\\?=8000 --env=ENDPOINT\\?=foo.com|--env=ENDPOINT\\?=foo.com --env=PORT\\?=8000) /node server.js",
			"/node server.js", map[string]string{"PORT": "8000", "ENDPOINT": "foo.com"}, true,
			[]string{"--env=PORT?=8000", "--env=ENDPOINT?=foo.com"},
			"",
		},
	}
```

Comment (miha-plesko) [inline test/runtime/runtime_test.go]: I understand now. So I've modified so that we use list instead of map. I like it how now test is more readable. Also, the checker is now more general (supports validating any `PREFIX_LIST + BASE` string). We should probably find a better name for it, but I can't think of one ATM. Any suggestion?
